### DISCUSSION TITLE
Quarto with julia

### DISCUSSION THREAD

**Discussion by ivanuricardo** [quarto-dev/quarto-nvim#97] [Q&A] [answered]
Hello,
I have been using the quarto nvim set up with lazyvim now. But now, I get an error message whenever I open a quarto document with julia code. However, once I enter past the error, everything works ok. I can quarto preview, I can run julia chunks, etc. Is there any way to get rid of this error message on the opening of a quarto document? This error seems to only come up when the document has code chunks.
```
Error detected while processing BufReadPost Autocommands for "*":
Error executing lua callback: /usr/share/nvim/runtime/filetype.lua:24: Error executing lua: /usr/share/nvim/run
time/filetype.lua:25: BufReadPost Autocommands for "*"..FileType Autocommands for "quarto"..FileType Autocomman
ds for "*"..function <SNR>1_LoadFTPlugin[19]..script /home/ivan/.local/share/nvim/lazy/quarto-nvim/ftplugin/qua
rto.lua: Vim(runtime):E5113: Error while calling lua chunk: ...n/.local/share/nvim/lazy/otter.nvim/lua/otter/ke
eper.lua:265: Invalid 'group': 'lspconfig'
stack traceback:
        [C]: in function 'nvim_get_autocmds'
        ...n/.local/share/nvim/lazy/otter.nvim/lua/otter/keeper.lua:265: in function 'activate'
        ...n/.local/share/nvim/lazy/quarto-nvim/lua/quarto/init.lua:140: in function 'activate'
        ...n/.local/share/nvim/lazy/quarto-nvim/ftplugin/quarto.lua:20: in main chunk
        [C]: in function 'nvim_exec_autocmds'
        ...hare/nvim/lazy/lazy.nvim/lua/lazy/core/handler/event.lua:157: in function <...hare/nvim/lazy/lazy.nv
im/lua/lazy/core/handler/event.lua:156>
        [C]: in function 'xpcall'
        .../.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/util.lua:113: in function 'try'
        ...hare/nvim/lazy/lazy.nvim/lua/lazy/core/handler/event.lua:156: in function 'trigger'
        ...hare/nvim/lazy/lazy.nvim/lua/lazy/core/handler/event.lua:87: in function <...hare/nvim/lazy/lazy.nvi
m/lua/lazy/core/handler/event.lua:72>
        [C]: in function 'nvim_cmd'
        /usr/share/nvim/runtime/filetype.lua:25: in function </usr/share/nvim/runtime/filetype.lua:24>
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:24: in function </usr/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_cmd'
        /usr/share/nvim/runtime/filetype.lua:25: in function </usr/share/nvim/runtime/filetype.lua:24>
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:24: in function </usr/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:24: in function </usr/share/nvim/runtime/filetype.lua:10>
```

**Comment by jmbuhr** [comment]
I'm also currently having problems starting the julia language server. Do you also have it exit with an error when editing a regular julia file?

**Reply by ivanuricardo** [reply]
Actually no. I can use a regular julia file fine. But I found that this problem occurs also with python code chunks and r code chunks...

**Reply by jmbuhr** [reply]
Do you have the R, python and julia language servers configured and installed? And you get e.g. code completion and documentation from those in plain R, py, jl documents?

**Reply by ivanuricardo** [reply]
I get documentation for pure python documents, so I can do Shift+k and find the documentation for e.g. dict. Shift+k won't work in julia, but I can type latex characters and run my code in julia files. In R files, nothing seems to work. It seems like the snippets work in R but nothing else.

**Reply by ivanuricardo** [reply]
In R, doing shift+k gives me documentation for the `apply` function, but not for the `matrix` function

**Reply by jmbuhr** [reply]
So the julia language server is indeed not set up.

**Reply by jmbuhr** [reply]
But it sounds like you are not actively using those anyways? You can turn off lsp features for quarto-nvim in the config: https://github.com/quarto-dev/quarto-nvim?tab=readme-ov-file#configure

**Reply by jmbuhr** [reply, marked as answer]
There may also have been a change in how LazyVim handles language servers. I updated https://github.com/jmbuhr/lazyvim-starter-for-quarto to better accomodate distributions that do their own thing.

**Reply by ivanuricardo** [reply]
Beautiful!! It works :D

**Reply by hashfwu** [reply]
Hi, thank you very much for the plugin, it is truly beautiful, and it has worked for me. Well, I'm just here to say something that might help new users a bit, I don't know. Anyway, regarding the Julia language, I also had a problem, I couldn't run it, I was getting:

```
No code chunks found for the current language, which is detected based on the current code block. Is your cursor in a code block?
```

What happened exactly was this problem where I didn't have the language server installed. What I did was:

```
:TSInstall julia
```

And then it recognizes the Julia code blocks.

Now I am using `quarto-nvim` + `molten-nvim` + `lazyvim`.

At the moment all scripts run in a single kernel, either Python or Julia, I can't run bash scripts, I suppose it's because of Molten that connects to a kernel, so I guess I'll fix that.

The way I made the Julia kernel visible was by doing:

```
Pkg.add("IJulia")
```

And nothing else, I just wanted to thank you for the great plugin and leave this comment in case it is useful to some new user (like me).

https://github.com/user-attachments/assets/cd00774a-a073-43d9-b9c1-df0e772c7d8a

**Reply by jmbuhr** [reply]
Thanks, that's great to hear! :)
Small correction: TSInstall stands for Tree Sitter, which is not the language server, but rather the system in Neovim that handles parsing filetypes, such that it knows e.g. where what language begins and what should be highlighted as a keyword.
